File: plugins/zego-assistant/skills/server-api-integration/examples/Signature.cs
using System;
using System.Security.Cryptography;
using System.Text;

namespace Zego.Util
{
    public static class ZegoSignature
    {
        // Signature = md5(AppId + SignatureNonce + ServerSecret + Timestamp)
        public static string GenerateSignature(uint appId, string signatureNonce, string serverSecret, long timestamp)
        {
            string str = $"{appId}{signatureNonce}{serverSecret}{timestamp}";
            using (MD5 md5 = MD5.Create())
            {
                byte[] hash = md5.ComputeHash(Encoding.UTF8.GetBytes(str));
                return BytesToHex(hash);
            }
        }

        // Generate a 16-character hex string (8 bytes)
        public static string GenerateSignatureNonce()
        {
            byte[] bytes = new byte[8];
            using (RandomNumberGenerator rng = RandomNumberGenerator.Create())
            {
                rng.GetBytes(bytes);
            }
            return BytesToHex(bytes);
        }

        private static string BytesToHex(byte[] bytes)
        {
            StringBuilder sb = new StringBuilder();
            foreach (byte b in bytes)
            {
                sb.Append(b.ToString("x2"));
            }
            return sb.ToString();
        }
    }
}

File: plugins/zego-assistant/skills/server-api-integration/examples/Signature.java
package com.zego.util;

import java.security.MessageDigest;
import java.security.SecureRandom;

public class ZegoSignature {

    // Signature = md5(AppId + SignatureNonce + ServerSecret + Timestamp)
    public static String generateSignature(long appId, String signatureNonce, String serverSecret, long timestamp) {
        String str = String.valueOf(appId) + signatureNonce + serverSecret + String.valueOf(timestamp);
        try {
            MessageDigest md = MessageDigest.getInstance("MD5");
            byte[] bytes = md.digest(str.getBytes("utf-8"));
            return bytesToHex(bytes);
        } catch (Exception e) {
            throw new RuntimeException("Failed to generate signature", e);
        }
    }

    // Generate a 16-character hex string (8 bytes)
    public static String generateSignatureNonce() {
        byte[] bytes = new byte[8];
        new SecureRandom().nextBytes(bytes);
        return bytesToHex(bytes);
    }

    private static String bytesToHex(byte[] bytes) {
        StringBuilder sb = new StringBuilder();
        for (byte b : bytes) {
            int digital = b & 0xFF;
            if (digital < 16) {
                sb.append("0");
            }
            sb.append(Integer.toHexString(digital));
        }
        return sb.toString();
    }
}
